esi/lease/v1/_proxy.py

```diff
@@ -0,0 +1,242 @@
+
+            * ``status``: Only return offers with the specified status.
+            * ``start_time``: Only return those with the specified ``start_time``
+              or an empty set if not found.
+            * ``end_time``: Only return those with the specified ``end_time``
+              or an empty set if not found.
+            * ``available_start_time``:  Only return those with the
+                specified ``available_start_time`` or an empty set if not found.
+            * ``available_end_time``:  Only return those with the
+              specified ``available_start_time`` or an empty set if not found.
+            * ``project_id``:only return the ones associated with this specific
+              project UUID.
+            * ``node_type``:only return the ones associated with this specific
+              node type.
+            * ``resource_uuid``:only return the ones associated with this specific
+              resource uuid.
+            * ``resource_class``:only return the ones associated with this specific
+              resource class.
+
+        :returns: A generator of offer instances.
+        """
+        return _offer.Offer.list(self, details=details, **query)
+
+    def create_offer(self, **attrs):
+        """Create a new offer from attributes.
+
+        :param dict attrs: Keyword arguments that will be used to create a
+            :class:`~esi_leap.v1.offer.Offer`.
+
+        :returns: The results of offer creation.
+        :rtype: :class:`~esi_leap.v1.offer.Offer`.
+        """
+        return self._create(_offer.Offer, **attrs)
+
+    def get_offer(self, offer, fields=None):
+        """Get a specific offer.
+
+        :param offer: The value can be the ID of an offer or a
+            :class:`~esi_leap.v1.offer.Offer` instance.
+        :param fields: Limit the resource fields to fetch.
+
+        :returns: One :class:`~esi_leap.v1.offer.Offer`
+        :raises: :class:`~openstack.exceptions.ResourceNotFound` when no
+            offer matching the name or ID could be found.
+        """
+        return self._get_with_fields(_offer.Offer, offer, fields=fields)
+
+    def delete_offer(self, offer, ignore_missing=True):
+        """Delete an offer.
+
+        :param offer: The value can be either the ID of an offer or
+            a :class:`~esi_leap.v1.offer.Offer` instance.
+        :param bool ignore_missing: When set to ``False``, an exception
+            :class:`~openstack.exceptions.ResourceNotFound` will be raised
+            when the offer could not be found. When set to ``True``, no
+            exception will be raised when attempting to delete a non-existent
+            offer.
+
+        :returns: The instance of the offer which was deleted.
+        :rtype: :class:`~esi_leap.v1.offer.Offer`.
+        """
+        return self._delete(_offer.Offer, offer, ignore_missing=ignore_missing)
+
+    def claim_offer(self, offer, new_lease, **attrs):
```

Comment:
When you call `claim_offer` above you don't pass in `new_lease`. I kinda suspect that `**attrs` is supposed to represent the optional new lease fields... ?

Reply:
yeah, new_lease actually includes **attr fields, I'll make this part clean